src/main/java/core/basesyntax/service/impl/SupplyAction.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ActionHandler;
+
+public class SupplyAction implements ActionHandler {
+    public void apply(String fruit, int quantity) {
+        Storage.addFruit(fruit, quantity);
+    }
```

Comment:
The `apply` method should not directly interact with the `Storage` class. Instead, it would be better to work with a service layer that interacts with the storage. This will make your code more in line with the SOLID principles, particularly the Single Responsibility Principle, as it abstracts the storage manipulation away from the action handling.